### PR TITLE
Proof of Concept: Ghost Ships

### PR DESCRIPTION
**Pull request by Karmic-Skink**
## About The Pull Request
This PR introduces the concept of ghost role ships, that *should* add a small amount of natural asymmetrical PvP to NSV13 rounds by adding support for ghosts to inhabit overmap entities directly.

NOTE: Currently set for 15 and 10 qualifying player requirements.

Additionally; this PR splits and has a light rework of overmap control schemes (see embedded help in-game for details), re-paths and better assigns overmap gamestate controls to the main control panel.

Todo:
- [x] Retool system for open candidate selection to use the PollGhost systems like I should have done from the start
- [x] Update pollcandidate to use ship name rather than object reference
- [x] Reallow small craft to free mouse
- [x] Reassess method for generating control input help messages when interacting with "pilot" object
- [ ] Fix hail commands
- [ ] Investigate re-implementing AMS and FLAK on ghost ships

## Why It's Good For The Game
Adds additional depth to overmap fights.
Addresses awful Q & E to turn buttons amongst other UI issues.

## Changelog
:cl:
refactor: Refactored flight controls by splitting into small and large craft control schemes
refactor: Defaulted small craft into using laser steering mode (note: keyboard steering unavailable)
refactor: Defaulted large craft into using keyboard steering mode
refactor: Refactored keyboard steering
tweak: Updated the ship help verb to accurately list controls
tweak: Set ship help verb to fire whenever a mob starts piloting 
add: Added ghost ships code
add: Added ghost ship encounters to fleet battles
add: Added support for tactical consoles in ghost ships
admin: Added various spawn options for ghost ships
admin: Retooled NSV13 specific ghost tools onto the overmap gamemode control panel
/:cl:

### CODE REVIEW

**Comment by skigear** [comment]
Does this PR remove strafing on all ships?

**Comment by Bokkiewokkie** [comment]
> Does this PR remove strafing on all ships?

image
No

**Comment by Karmic-Skink** [comment]
> 

They couldn't strafe anyway.

**Comment by Karmic-Skink** [comment]
> > Does this PR remove strafing on all ships?
> 
> image No only big ones.

They couldn't strafe anyway.

**Comment by pupertino** [comment]
I like the idea of ghost ship roles but I don't like the idea of dumbing down the cap ship controls by removing IAS toggle. You will no longer be able to use boost to do those sharp turns that are basically essential for Galactica to not be completely useless. And you might as well remove the frontal phasers from whisp if this goes through.

**Comment by pupertino** [comment]
Or this is about just removing sideways thrusters? Those were never needed for strafing.

**Comment by skigear** [comment]
> Or this is about just removing sideways thrusters? Those were never needed for strafing.

Yeah this is about sideways thrusters which can be used for fine adjustment and dodging things.

> > > Does this PR remove strafing on all ships?
> > 
> > 
> > image No only big ones.
> 
> They couldn't strafe anyway.

Which ships are the big ones? iirc ships like the Atlas, Shrike and Eclipse are capable of moving side to side alright and it can be very useful. I feel like this would lock people into using IAS a lot more because of the loss of that control.

**Comment by pupertino** [comment]
> > Or this is about just removing sideways thrusters? Those were never needed for strafing.
> 
> Yeah this is about sideways thrusters which can be used for fine adjustment and dodging things.
OK it's probably fine then and won't change things too much. Might make controlling cap ships a slight bit harder but not impossibly so.

**Comment by Karmic-Skink** [comment]
> I like the idea of ghost ship roles but I don't like the idea of dumbing down the cap ship controls by removing IAS toggle. You will no longer be able to use boost to do those sharp turns that are basically essential for Galactica to not be completely useless. And you might as well remove the frontal phasers from whisp if this goes through.

You need to go read what this actually does.

File: nsv13/code/modules/overmap/overmap_ghosts.dm [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+					var/mob/dead/observer/C = pick(candidates)
+					target_ghost = C
+				else
+					return
+			if("Choose")
+				target_ghost = input(usr, "Select player to pilot ghost ship:", "Select Player") as null|anything in GLOB.clients
+
+		ghost_ship(target_ghost)
+		message_admins("[key_name_admin(usr)] has ghost shiped [src.name]!")
+		log_admin("[key_name_admin(usr)] has ghost shiped [src.name]!")
```
Comment (Bokkiewokkie) [inline nsv13/code/modules/overmap/overmap_ghosts.dm:30]:
```suggestion
		message_admins("[key_name_admin(usr)] has ghost shipped [src.name]!")
		log_admin("[key_name_admin(usr)] has ghost shipped [src.name]!")
```
shipped and tested, alternatively:
```suggestion
		message_admins("[key_name_admin(usr)] has turned [src.name] into a ghost ship!")
		log_admin("[key_name_admin(usr)] has turned [src.name] into a ghost ship!")
```

**Review by covertcorvid — APPROVED** [review]

**Pull request merged**